extensions/omni_httpd/src/routing/make_router.sql

```diff
@@ -0,0 +1,65 @@
+CREATE OR REPLACE FUNCTION make_router(
+    router_name name,
+    router_type name,
+    route_regexp text DEFAULT '.*',
+    method text DEFAULT 'GET',
+    host_regexp text DEFAULT '.*'
+) RETURNS void AS $$
+DECLARE
+    type_columns text[];
+    capture_count int;
+    function_body text;
+    type_definition text;
+BEGIN
+    -- Get the columns of the composite type
+    SELECT array_agg(attname::text)
+    INTO type_columns
+    FROM pg_attribute
+    WHERE attrelid = router_type::regclass
+    AND attnum > 0
+    AND NOT attisdropped;
+
+    -- Count capturing groups in route_regexp
+    SELECT regexp_matches(route_regexp, '\((?!\?:)', 'g')
```

Comment:
Check out `omni_regex`, it has support for [named capture groups](https://docs.omnigres.org/omni_regex/regex/#named-catpure-groups) which will make them align with column names much smoother.